src/components/movieForm/index.tsx

```diff
@@ -48,58 +79,99 @@ const MovieForm: React.FC<IMovieFormProps> = ({
     setSelectedOptions(data.selectedOptions);
     setValue(data.selectedOptions.join(', '));
   };
-  const [value, setValue] = React.useState(initialMovieInfo?.genres.join(', '));
+  const [value, setValue] = React.useState(
+    initialMovieInfo?.genres.join(', ') ?? ''
+  );
 
   const optionList = genres.map((option) => {
     return (
-      <Option className={styles.option} key={option}>
+      <Option
+        className={styles.option}
+        key={option}
+        value={option}
+        data-testid='movie-form-genre-option'
+      >
         {option}
       </Option>
     );
   });
 
   return (
-    <form className={styles.root} onSubmit={handleSubmit} data-testid='form'>
+    <form
+      className={styles.root}
+      onSubmit={handleSubmit(handleFormSubmit)}
+      onReset={() => {
+        setSelectedOptions(initialMovieInfo?.genres ?? []);
+        setValue(initialMovieInfo?.genres.join(', ') ?? '');
```

Comment:
You need to pass proper defaultValues to useForm and all reset action will be done by form lib internally. You just need to call reset(). https://react-hook-form.com/api/useform/reset/